gems/aws-sdk-core/spec/aws-sdk-core/middleware/checksums_spec.rb

```diff
@@ -0,0 +1,234 @@
+# frozen_string_literal: true
+
+require_relative '../../spec_helper'
+require 'hearth'
+
+module AWS::SDK::Core
+  module Middleware
+    Input = ::Struct.new(
+      :request_algorithm,
+      :request_validation
+    ) do
+      include Hearth::Structure
+    end
+
+    describe Checksum do
+      let(:output) { Hearth::Output.new }
+      let(:app) { double('app', call: output) }
+
+      let(:request_algorithm_member) { :request_algorithm }
+      let(:request_checksum_required) { false }
+      let(:streaming) { false }
+      let(:signed_streaming) { false }
+      let(:require_decoded_content_length) { false }
+      let(:request_validation_mode_member) { :request_validation }
+      let(:response_algorithms) { ['SHA256'] }
+
+      subject do
+        Checksum.new(
+          app,
+          request_algorithm_member: request_algorithm_member,
+          request_checksum_required: request_checksum_required,
+          streaming: streaming,
+          signed_streaming: signed_streaming,
+          require_decoded_content_length: require_decoded_content_length,
+          request_validation_mode_member: request_validation_mode_member,
+          response_algorithms: response_algorithms
+        )
+      end
+
+      describe '#call' do
+        let(:input) { Input.new }
+        let(:request_body) { StringIO.new('Hello World') }
+        let(:response_body) { StringIO.new }
+
+        let(:request) do
+          Hearth::HTTP::Request.new(
+            body: request_body
+          )
+        end
+
+        let(:response) do
+          Hearth::HTTP::Response.new(
+            body: response_body
+          )
+        end
+
+        let(:context) do
+          Hearth::Context.new(
+            request: request,
+            response: response
+          )
+        end
+
+        context 'request checksum not set' do
+          let(:input) { Input.new }
+
+          it 'does not compute a checksum' do
+            subject.call(input, context)
+            expect(request.headers.to_h.keys
```

Comment:
headers.keys should work here. But also, I think there's an rspec matcher for not including

Reply:
I couldn't find one that would compare the keys using a prefix (since we need to just check that it does not include any of the checksum headers, which all start with the same prefix)